packages/iris-grid/src/IrisGridTableModel.ts

```diff
@@ -78,21 +80,32 @@ class IrisGridTableModel
     return this.table.applyCustomColumns != null;
   }
 
+  getMemoizedKeyColumnSet = memoize(
+    (inputTableKeys?: readonly string[]) =>
+      new Set(inputTableKeys ?? EMPTY_ARRAY)
+  );
+
+  get keyColumnSet(): Set<ColumnName> {
+    return this.getMemoizedKeyColumnSet(this.inputTable?.keys ?? EMPTY_ARRAY);
```

Comment:
Don't need to fallback here, since the `getMemoizedKeyColumnSet` can handle `undefined` being passed in
```suggestion
    return this.getMemoizedKeyColumnSet(this.inputTable?.keys);
```